flink-runtime/src/main/java/org/apache/flink/runtime/state/LocalRecoveryConfig.java

```diff
@@ -44,7 +41,6 @@ public boolean isLocalRecoveryEnabled() {
         return localRecoveryEnabled;
     }
 
-    @Nonnull
     public LocalRecoveryDirectoryProvider getLocalStateDirectoryProvider() {
```

Comment:
Would it make sense to change the signature to `Optional<LocalRecoveryDirectoryProvider> getLocalStateDirectoryProvider()` to make sure the change is properly reflected in all places?

Other option would be adding some explicit exception when getting a null provider.

Reply:
The signature of optional may be a better option. We could use orElseThrow to throw exceptions although it should not happen when enabled task local recovery.